### PR TITLE
feat: implement characters tab

### LINKED ISSUE #6: Characters tab

### PR DESCRIPTION
**Pull request by Moltenship**

### CODE REVIEW

**Review by moetazaneta — APPROVED** [review]

**Review by prekel — CHANGES_REQUESTED** [review]
image

File: src/api/services/characterService.ts [hunk 1]
```diff
@@ -6,15 +6,37 @@ import { DataWrapper } from '../dtos/dataWrapper';
...
+ * Fetches all vn characters by vn id.
  * @param vnId Visual novel id.
- * TODO: Add support for fetching more.
  */
-export const fetchCharacters = async(vnId: string): Promise<Character[]> => {
-  const { data } = await http.post<DataWrapper<CharacterDto>>(
-    ApiUrls.Vndb,
-    `get character basic,details,meas,voiced,traits,vns (vn = ${vnId})`,
-  );
+export const fetchCharactersByVnId = async(vnId: string): Promise<Character[]> => {
```
Comment (NightOfNights) [inline src/api/services/characterService.ts]: Use `VisualNovel['id']` instead of `string`

File: src/api/services/characterService.ts [hunk 2]
```diff
@@ -6,15 +6,37 @@ import { DataWrapper } from '../dtos/dataWrapper';
...
+    charactersChunk.push(charactersData.data.items);
+
+    if (charactersData.data.more) {
+      currentPage += 1;
+    } else {
+      hasToFetchMore = false;
+    }
+  }
+
+  return charactersChunk.flatMap(characterChunk => characterChunk.map(dto => characterFromDto(dto)));
```
Comment (NightOfNights) [inline src/api/services/characterService.ts]: Copy paste of https://github.com/vndbre/vndbre/blob/develop/src/api/services/releaseService.ts

File: src/features/VisualNovel/pages/CharactersPage/CharactersPage.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  );
+
+  return (
+    <ContentWrapper isLoading={isCharactersLoading} error={charactersError}>
+      {data && data?.length > 0 ? (
+        <Accordion defaultIndex={[0]} allowMultiple>
+          {groupedCharacterElements}
+        </Accordion>
+
+      ) : <Text>No characters :(</Text>}
```
Comment (NightOfNights) [inline src/features/VisualNovel/pages/CharactersPage/CharactersPage.tsx]: Is `:(` thing intended by designs?

**Review by NightOfNights — CHANGES_REQUESTED** [review]

**Review by prekel — CHANGES_REQUESTED** [review]
> `src/features/VisualNovel/pages/MediaPage/MediaPage.tsx(15,58): error TS2345: Argument of type 'string' is not assignable to parameter of type 'number'.`

Please fix compilation error.

**Review by prekel — CHANGES_REQUESTED** [review]
Please resolve conflicts.

**Review by NightOfNights — APPROVED** [review]

**Pull request merged**